### ISSUE TITLE
Error `/bin/sh: Argument list too long` when doing `$ make install`

### ISSUE THREAD

**Issue by benediktahrens** [UniMath/UniMath#1757]
I am seeing this error on Debian:
```
$ make install
ulimit -v unlimited ; make -f build/CoqMakefile.make all
make[1]: Entering directory '/home/ahrens/github/UniMath/UniMath'
make[2]: Nothing to be done for 'real-all'.
make[1]: Leaving directory '/home/ahrens/github/UniMath/UniMath'
ulimit -v unlimited ; make -f build/CoqMakefile.make install
make[1]: Entering directory '/home/ahrens/github/UniMath/UniMath'
make[1]: /bin/sh: Argument list too long
make[1]: *** [build/CoqMakefile.make:598: install] Error 127
make[1]: Leaving directory '/home/ahrens/github/UniMath/UniMath'
make: *** [Makefile:99: install] Error 2
```
Do you know of any way around this?

**Comment by nmvdw** [comment]
Maybe relevant info: I don't get this error on Mac OS (so, it might be Debian related).

**Comment by benediktahrens** [comment]
Is the solution here: https://stackoverflow.com/questions/11475221/why-do-i-get-bin-sh-argument-list-too-long-when-passing-quoted-arguments ?

**Comment by dwarfmaster** [comment]
I have the same error on NixOS, so it seems to be linux specific.

**Comment by benediktahrens** [comment]
@dwarfmaster : what does the output of `$ readlink -f /bin/sh` show for you? For me it shows `/usr/bin/dash`.

**Comment by dwarfmaster** [comment]
Since I am trying to build it using nix, it doesn't use `/bin/sh` but `/nix/store/8fv91097mbh5049i9rglc73dx6kjg3qk-bash-5.2-p15/bin/bash`, which is `bash`. If the stackoverflow thread you have linked is to be trusted, it should be a limitation in the linux kernel, not in the shell itself.

**Comment by benediktahrens** [comment]
The Coq developers have encountered the same problem with UniMath recently: https://coq.zulipchat.com/#narrow/stream/250632-Coq-Platform-devs-.26-users/topic/Strange.20command.20length.20limit.20on.20Linux

**Comment by benediktahrens** [comment]
A fix seems to be implemented here: https://github.com/coq/coq/pull/17697 The files to be installed are being written to a file instead of passed in a list.

**Comment by dwarfmaster** [comment]
Does this mean the only way to use unimath now is to use coq's master branch ? Because this is a bit problematic for plugin developpers...

**Comment by benediktahrens** [comment]
If I understand https://github.com/coq/coq/pull/17697 correctly, specifically, this message
"moved this from Request 8.18+rc1 inclusion to Shipped in 8.18+rc1 in [Coq 8.18](https://github.com/coq/coq/projects/57#card-89747440)"
then the fix will be shipped in Coq 8.18, which could be out soon.

**Comment by DanGrayson** [comment]
I can't reproduce this under Debian 11.  Benedikt, did you install bash, or did you create a link named "bash" to another shell?

**Comment by benediktahrens** [comment]
My `/bin/sh` points to "dash". But note that the Coq developers noticed this problem before us (specifically, the Coq Platform developers) and a fix has been prepared that will be included in Coq 8.18. The fix is to write the files to be installed into a file, rather than passing a long string around.

**Comment by m-lindgren** [comment]
> Does this mean the only way to use unimath now is to use coq's master branch ?

No, there's also the option to use [dune](https://dune.build/) to build UniMath. `dune install` shouldn't be affected by this issue.

**Comment by dwarfmaster** [comment]
If so it may be interesting to use dune in the opam file, because right now installing unimath with opam fails due to this error.

**Comment by arnoudvanderleer** [comment]
@benediktahrens Has this been solved in the meantime?

**Comment by benediktahrens** [comment]
@arnoudvanderleer : I believe that the fix for #1819 - which you implemented, thanks a lot - also fixed this issue. Closing.
